sympy/solvers/solveset.py

```diff
@@ -931,6 +932,26 @@ def solveset(f, symbol=None, domain=S.Complexes):
             raise NotImplementedError(filldedent('''
                 relationship between value and 0 is unknown: %s''' % b))
 
+    rep = []
+    if f.has(Abs):
+        # masking off the Abs() part from being rewritten into Piecewise
+        from sympy import numbered_symbols
+        e = [a for a in f.atoms(Abs) if symbol in a.free_symbols]
+        rep = numbered_symbols("t", cls=Dummy)
+        rep = [next(rep) for _ in range(len(e))]
+        newf = f
```

Comment:
You can define a `reps` and `inv_reps` as:

```
reps = dict(zip(e, numered_symbols("t", cls=Dummy)))
inv_reps = [(v, k) for k, v in reps.items()]
```

Restoration should be done with `xreplace` since the keys will be symbols: `f = f.rewrite(Piecewise).xreplace(inv_reps)`

Reply:
for example...

```diff
diff --git a/sympy/solvers/solveset.py b/sympy/solvers/solveset.py
index 85b4d7d..9e22c9b 100644
--- a/sympy/solvers/solveset.py
+++ b/sympy/solvers/solveset.py
@@ -32,6 +32,7 @@
 from sympy.solvers.polysys import solve_poly_system
 from sympy.solvers.inequalities import solve_univariate_inequality
 from sympy.utilities import filldedent
+from sympy.utilities.iterables import numbered_symbols
 from sympy.calculus.util import periodicity, continuous_domain
 from sympy.core.compatibility import ordered, default_sort_key, is_sequence
 from sympy.functions.elementary.piecewise import Piecewise
@@ -932,25 +933,18 @@ def solveset(f, symbol=None, domain=S.Complexes):
             raise NotImplementedError(filldedent('''
                 relationship between value and 0 is unknown: %s''' % b))

-    rep = []
-    if f.has(Abs):
-        # masking off the Abs() part from being rewritten into Piecewise
-        from sympy import numbered_symbols
-        e = [a for a in f.atoms(Abs) if symbol in a.free_symbols]
-        rep = numbered_symbols("t", cls=Dummy)
-        rep = [next(rep) for _ in range(len(e))]
-        newf = f
-        rep_dict = {}
-        for r in e:
-            newf =  newf.replace(r, rep[e.index(r)])
-            rep_dict[rep[e.index(r)]] = r
-        f = newf
-
+    reps = dict(zip(
+        # do larger expressions first in case they contain
+        # other Abs
+        reversed(list(ordered(
+            [i for i in f.atoms(Abs) if symbol in i.free_symbols]))),
+        numbered_symbols("t", cls=Dummy)))
+    for k in reps:
+        f = f.replace(k, reps[k])
     f = f.rewrite(Piecewise)
-
-    for i in rep:
-        # replacing back the masked off part
-        f = f.replace(i, rep_dict[i])
+    if reps:
+        ireps = dict([(v, k) for k, v in reps.items()])
+        f = f.xreplace(ireps)

     if isinstance(f, Eq):
         from sympy.core import Add
```